x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -1,17 +1,119 @@
 package simulation
 
-// func RandomMsgCreateConcentratedPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*clmodeltypes.MsgCreateConcentratedPool, error) {
-// 	return nil, nil
-// }
+import (
+	"errors"
+	"fmt"
+	"math/rand"
 
-// func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgCreatePosition, error) {
-// 	return nil, nil
-// }
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	legacysimulationtype "github.com/cosmos/cosmos-sdk/types/simulation"
+
+	osmosimtypes "github.com/osmosis-labs/osmosis/v14/simulation/simtypes"
+	clkeeper "github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity"
+	clmodeltypes "github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity/model"
+	cltypes "github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity/types"
+)
+
+var PoolCreationFee = sdk.NewInt64Coin("stake", 10_000_000)
+
+func RandomMsgCreateConcentratedPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*clmodeltypes.MsgCreateConcentratedPool, error) {
+	// generate random values from -13 to 1 (accepted range: -12 to -1)
+	exponentAtPriceOne := sdk.NewInt(rand.Int63n(1+13) - 13)
+	authorizedTickSpacing := []uint64{1, 10, 60, 200}
+
+	// find an address with two or more distinct denoms in their wallet
+	sender, senderExists := sim.RandomSimAccountWithConstraint(createPoolRestriction(k, sim, ctx))
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with two different denoms & pool creation fee exists")
+	}
+
+	// generate 3 coins, use 2 to create pool and 1 for fees. ["stake" denom - contiains invalid metadata]
```

Comment:
["stake" denom - contains invalid metadata] - what does this mean?

```suggestion
	// generate 3 coins, use 2 to create pool and 1 for fees. ["stake" denom - contains invalid metadata]
```